components/event-sources/reconciler/httpsource/reconcile.go

```diff
@@ -242,6 +260,25 @@ func (r *Reconciler) syncKnService(src *sourcesv1alpha1.HTTPSource,
 	return ksvc, nil
 }
 
+// syncChannel synchronizes the desired state of a Channel against its current
+// state in the running cluster.
+func (r *Reconciler) syncChannel(src *sourcesv1alpha1.HTTPSource,
+	currentCh, desiredCh *messagingv1alpha1.Channel) (*messagingv1alpha1.Channel, error) {
+
+	if objects.Semantic.DeepEqual(currentCh, desiredCh) {
+		return currentCh, nil
+	}
+
+	ch, err := r.messagingClient.Channels(currentCh.Namespace).Update(desiredCh)
+	if err != nil {
+		r.eventWarn(src, failedUpdateReason, "Update failed for Channel %q", ch.Name)
```

Comment:
At this point, `ch` could be nil.

Spotted the same for line **255**, `ksvc` could be nil as well.

Reply:
Nice catch, fixed